iio/iio_ad713x/iio_ad713x.c

```diff
@@ -0,0 +1,305 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <inttypes.h>
+#include <string.h>
+#include <errno.h>
+#include <stdlib.h>
+#include "error.h"
+#include "xml.h"
+#include "util.h"
+#include "iio.h"
+#include "iio_ad713x.h"
+#include "spi_engine_core.h"
+#include "spi_engine_offload.h"
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+static const char ad713x_xml[] =
+	"<device id=\"adc\" name=\"adc\" >"
+	"<channel id=\"voltage0\" type=\"input\" >"
+	"<scan-element index=\"0\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage1\" type=\"input\" >"
+	"<scan-element index=\"1\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage2\" type=\"input\" >"
+	"<scan-element index=\"2\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage3\" type=\"input\" >"
+	"<scan-element index=\"3\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage4\" type=\"input\" >"
+	"<scan-element index=\"4\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage5\" type=\"input\" >"
+	"<scan-element index=\"5\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage6\" type=\"input\" >"
+	"<scan-element index=\"6\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"<channel id=\"voltage7\" type=\"input\" >"
+	"<scan-element index=\"7\" format=\"le:S12/16&gt;&gt;0\" />"
+	"</channel>"
+	"</device>"
+	;
+
+/**
+ * @brief iio_axi_dac_delete_device() - Delete iio_device.
+ * @param iio_device - Structure describing a device, channels and attributes.
+ * return SUCCESS in case of success or negative value otherwise.
+ */
+ssize_t iio_ad713x_delete_device(struct iio_device *iio_device)
+{
+	if(iio_device)
+		free(iio_device);
+
+	return SUCCESS;
+}
+
+/**
+ * @brief iio_axi_dac_create_device() - Create structure describing a device, channels and attributes.
+ * @param device - Device name.
+ * @param num_ch - Number of channels that the device has.
+ * return iio_device or NULL, in case of failure.
+ */
+struct iio_device *iio_ad713x_create_device(const char *device_name,
+		uint16_t num_ch)
+{
+	struct iio_device *iio_device;
+
+	iio_device = calloc(1, sizeof(struct iio_device));
+	if (!iio_device)
+		return NULL;
+
+	iio_device->name = device_name;
+	iio_device->num_ch = num_ch;
+	iio_device->attributes =
+		NULL; //iio_voltage_attributes; /* no device attribute */
```

Comment:
those comments :D

Reply:
also, is it normal not to define any IIO attributes ?

Reply:
It is OK. In the future if we want to read/write specific attributes, they should be added here. At the moment we only support capture, for this device.